src/App.tsx

```diff
@@ -56,7 +56,19 @@ const App: React.FC = () => {
 
   const updateStatus = useCallback(async () => {
     try {
-      setStatus(await fetchJSON<ScanStatusResponse>('/scan/status'))
+      const newStatus = await fetchJSON<ScanStatusResponse>('/scan/status')
+      setStatus((prevStatus) => {
+        const isSameStatus =
+          prevStatus.batches.filter((b) => b.endedAt).length ===
```

Comment:
there's a chance here that this will provide a false positive, as in the statuses look the same but they changed, specifically because the `endedAt` date sometimes is locked in before the last ballot has come in (longer delay that I've sometimes observed.)

I would recommend leaning more towards some false negatives --> slightly more aggressive refreshing, rather than not refreshing, cause not refreshing is bad news.

So I suggest just doing a string compare on the serialized old and new status.

Reply:
done